listenbrainz/webserver/views/api_tools.py

```diff
@@ -70,6 +70,16 @@ def _send_listens_to_queue(listen_type, listens):
             submit.append(listen)
 
     if submit:
+        # check if rabbitmq connection exists or not
+        # and if not then try to connect
+        if rabbitmq_connection._rabbitmq is None:
```

Comment:
Hmm. Using a _function from outside the module bugs me. I think we should add an accessor function here. Either that or always call init_rabbitmq_connection and have it do nothing if the connection is already established.